File: examples/csharp/ApibaraVehicleAuctionClient.cs
using System.Net.Http.Headers;
using System.Web;

public sealed class ApibaraVehicleAuctionClient
{
    private readonly HttpClient _httpClient;
    private readonly string _baseUrl;

    public ApibaraVehicleAuctionClient(string apiKey, string baseUrl = "https://apibara.tech/api/v1/vehicle-auction")
    {
        _baseUrl = baseUrl.TrimEnd('/');
        _httpClient = new HttpClient();
        _httpClient.DefaultRequestHeaders.Add("X-API-Key", apiKey);
        _httpClient.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
    }

    public async Task<string> GetAsync(string path, Dictionary<string, string?>? parameters = null)
    {
        var query = HttpUtility.ParseQueryString(string.Empty);

        foreach (var item in parameters ?? new Dictionary<string, string?>())
        {
            if (!string.IsNullOrWhiteSpace(item.Value))
            {
                query[item.Key] = item.Value;
            }
        }

        var url = _baseUrl + path + (query.Count > 0 ? "?" + query : "");
        var response = await _httpClient.GetAsync(url);
        var body = await response.Content.ReadAsStringAsync();

        if (!response.IsSuccessStatusCode)
        {
            throw new HttpRequestException($"Apibara API error {(int)response.StatusCode}: {body}");
        }

        return body;
    }
}

public static class Program
{
    public static async Task Main()
    {
        var apiKey = Environment.GetEnvironmentVariable("APIBARA_API_KEY")
            ?? throw new InvalidOperationException("Missing APIBARA_API_KEY");

        var client = new ApibaraVehicleAuctionClient(apiKey);

        var json = await client.GetAsync("/vehicles", new Dictionary<string, string?>
        {
            ["make"] = "Toyota",
            ["model"] = "Camry",
            ["year_from"] = "2018",
            ["lot_sub_status"] = "Open",
            ["per_page"] = "20"
        });

        Console.WriteLine(json);
    }
}

File: examples/swift/ApibaraVehicleAuctionClient.swift
import Foundation

struct ApibaraVehicleAuctionClient {
    let apiKey: String
    let baseURL: URL

    init(apiKey: String, baseURL: URL = URL(string: "https://apibara.tech/api/v1/vehicle-auction")!) {
        self.apiKey = apiKey
        self.baseURL = baseURL
    }

    func get(path: String, queryItems: [URLQueryItem] = []) async throws -> Data {
        var components = URLComponents(url: baseURL.appendingPathComponent(path), resolvingAgainstBaseURL: false)!
        components.queryItems = queryItems.isEmpty ? nil : queryItems

        var request = URLRequest(url: components.url!)
        request.httpMethod = "GET"
        request.setValue(apiKey, forHTTPHeaderField: "X-API-Key")
        request.setValue("application/json", forHTTPHeaderField: "Accept")

        let (data, response) = try await URLSession.shared.data(for: request)

        guard let httpResponse = response as? HTTPURLResponse,
              (200..<300).contains(httpResponse.statusCode) else {
            throw URLError(.badServerResponse)
        }

        return data
    }

    func searchVehicles() async throws -> Data {
        try await get(path: "vehicles", queryItems: [
            URLQueryItem(name: "make", value: "Toyota"),
            URLQueryItem(name: "model", value: "Camry"),
            URLQueryItem(name: "year_from", value: "2018"),
            URLQueryItem(name: "lot_sub_status", value: "Open"),
            URLQueryItem(name: "per_page", value: "20")
        ])
    }

    func getVehicleDetails(vinOrLot: String) async throws -> Data {
        try await get(path: "vehicles/\(vinOrLot)")
    }

    func getShipping(vinOrLot: String) async throws -> Data {
        try await get(path: "vehicles/\(vinOrLot)/shipping", queryItems: [
            URLQueryItem(name: "ports", value: "Miami,NY,LA")
        ])
    }
}
